### PR TITLE
feat: add webdav http methods

### PR DESCRIPTION
**Pull request by hungtcs**
Relat #9 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Do those method accept a body? Can you please test like the other methods are tested?


Those methods are not that common, I would not add the shorthands.

**Comment by hungtcs** [comment]
@mcollina Yes, some method receive a body of type xml sometimes, I will adjust some details later.

**Comment by hungtcs** [comment]
Looks like it's almost done, but before merging, I will write a simple WebDAV server for real world test.

File: lib/httpMethods.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+'use strict'
+
```
Comment (3imed-jaberi) [inline lib/httpMethods.js:2]: Maybe it's would be better to split the supported methods into 2 arrays. it helps when to like to provide an option endable/disable for [WebDAV](https://www.ibm.com/docs/en/i/7.1?topic=concepts-webdav) methods.

```js

const baseHtttpMethods = ['DELETE', 'GET', 'HEAD', 'PATCH', 'POST', 'PUT', 'OPTIONS']
const webDAVHtttpMethods = ['PROPFIND', 'PROPPATCH', 'MKCOL', 'COPY', 'MOVE', 'LOCK', 'UNLOCK', 'TRACE', 'SEARCH']

module.exports = {
  supportedMethods: [...baseHtttpMethods, ...webDAVHtttpMethods] // you can use concat or whatever!
}

```

**Review by 3imed-jaberi — APPROVED** [review]
sgtm +1 ✅

**Review by zqxul — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**